### PR TITLE
CI: Automate the updating of the list via XSS feed

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
This _tries_ to automate updating the least more frequently.

The `cron.yml` runs daily (or via workflow_dispatch), checks for new updates and update the ofac.sanctions.json file accordingly. If the file is updated, then it triggers a new release via `publish.yml`

### CODE REVIEW

File: scripts/get-latest-sanctions.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if (link in processedURLs) continue;
+
+    try {
+      const body = await fetch(link).then((x) => x.text());
+      const dom = new JSDOM(body);
+      const text =
+        dom.window.document.getElementById("block-ofac-content")?.textContent;
+      if (!text) throw new Error("Couldn't find the right div");
+
+      const RE = /ETH (0x[a-fA-F0-9]{40})/g;
```
Comment (Jesse-Sawa) [inline scripts/get-latest-sanctions.ts:29]: OFAC will repeat an address for different assets e.g. USDC, ETH, USDT, etc.

It's possible that  ETH isn't mentioned. 

Maybe remove that prefix. Most EVM chains us the 0x prefix with 20bytes so prob can just leave It as '(0x[a-fA-F0-9]{40})/g'

Comment (Jesse-Sawa) [inline scripts/get-latest-sanctions.ts:29]: OFAC will repeat an address for different assets e.g. USDC, ETH, USDT, etc.

It's possible that  ETH isn't mentioned. 

Maybe remove that prefix. Most EVM chains us the 0x prefix with 20bytes so prob can just leave It as '(0x[a-fA-F0-9]{40})/g'

File: scripts/get-latest-sanctions.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      const matches = text.matchAll(RE);
+      if (!matches) {
+        console.info("No ETH addresses were found");
+        processedURLs[link] = [];
```
Comment (Jesse-Sawa) [inline scripts/get-latest-sanctions.ts:33]: Check for duplicates? It will keep the list clean

**Review by Jesse-Sawa — APPROVED** [review]

**Comment by 0xarthurxyz** [comment]
I can roughly make sense of the new workflow by looking at the PR, but to avoid misunderstandings could you please describe the request flows and new design @nicolasbrugneaux (for future reference).

Rough sense I have:
- We have a Github Action that runs daily (defined in [`cron.yml`](https://github.com/celo-org/compliance/pull/1/files#diff-69b9772a7fe51bd7c367926862d89ca7d8949cc9832b3645e610a1ec5bfa1ddf)

I can see a couple new scripts [`export-sanctions.ts`](https://github.com/celo-org/compliance/pull/1/files#diff-5b778010df5fc76ce37104f02fd1169d754d18fcc99bf3afcc23ebade2c1d073), [`export-sanctions.ts`](https://github.com/celo-org/compliance/pull/1/files#diff-5b778010df5fc76ce37104f02fd1169d754d18fcc99bf3afcc23ebade2c1d073)

**Comment by nicolasbrugneaux** [comment]
Your summary is spot on @0xarthurxyz 🙌 . There's just a couple thing I'd like to address.

---

Firstly, I'd like to add is that the HTTP request isn't to the github repository but to a published ["Github Page"](https://pages.github.com/), published via the following workflow:
- https://github.com/celo-org/compliance/actions/runs/6596446191

This makes sure we don't rely on GH's infrastructure, URIs, filestructure, or cache-policies to serve the file adequately; it'll always be available at the [same URL ](https://celo-org.github.io/compliance/ofac.sanctions.json)

---

Another point, that I'd like to emphasize is the publishing process. A new version of the SDK and the json file is only published/deployed **if new addresses were found**. 
This is why there's a couple conditional build steps (`if: ${{ steps.bump_version.diff_flag.GIT_EXIT_CODE == 1 }}` in both yml files, this condition checks that the repository is dirty after fetching the addresses, meaning the json file was modified.

**Comment by 0xarthurxyz** [comment]
Brilliant, really appreciate the context @nicolasbrugneaux. 

I see that @Jesse-Sawa reviewed the changes is that sufficient from your perspective, or would you like a third pair of eyes from say @aaronmgdr ? 

This is a relatively high impact PR since it relates to compliance, so I'd err on the side of caution here, but you have a better hunch for the changes so happy to hear what you think.

In any case, this is brilliant, really appreciate it!

**Comment by aaronmgdr** [comment]
> Brilliant, really appreciate the context @nicolasbrugneaux.
> 
> I see that @Jesse-Sawa reviewed the changes is that sufficient from your perspective, or would you like a third pair of eyes from say @aaronmgdr before we put this to rest ?
> 
> This is a relatively high impact PR since it relates to compliance, so I'd err on the side of caution here, but you have a better hunch for the changes so happy to hear what you think.
> 
> In any case, this is brilliant, really appreciate it!

This looks good to me

**Pull request merged**